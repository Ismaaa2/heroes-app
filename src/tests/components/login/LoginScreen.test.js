import { mount } from "enzyme";
import { AuthContext } from "../../../auth/AuthContext";
import { LoginScreen } from "../../../components/login/LoginScreen";
import { types } from "../../../types/types";

describe("Name of the group", () => {
  const history = {
    push: jest.fn(),
    replace: jest.fn(),
    location: {},
    listen: jest.fn(),
    createHref: jest.fn(),
  };

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: "Ismael Bonilla",
    },
  };

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <LoginScreen history={history} />
    </AuthContext.Provider>
  );


  Storage.prototype.getItem = jest.fn();

  

  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de realizar el dispatch y la navegaciÃ³n", () => {
    const handleClick = wrapper.find("button").prop("onClick");

    handleClick()
    expect(contextValue.dispatch).toHaveBeenCalledTimes(1);
    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: {
        name: "Ismael",
      },
    });
    expect(history.replace).toHaveBeenCalledTimes(1);
    expect(history.replace).toHaveBeenCalledWith("/");

  });

  // test('OK', () => {

  //   const wrapper = mount(
  //       <AuthContext.Provider value={contextValue}>
  //         <LoginScreen history={history} />
  //       </AuthContext.Provider>
  //     );

  //   const handleClick = wrapper.find("button").prop("onClick");
  //   localStorage.setItem('lastPath', '/dc');
    
  //   handleClick()
  //   // expect(history.replace).toHaveBeenCalledWith("dc");
  // });
});





// test('dispatch and navigation',()=>{
//     const localStorageMock = (function() {
//               var store = {};
//               return {
//                 getItem: function(key) {
//                   return store[key];
//                 },
//                 setItem: function(key, value) {
//                   store[key] = value.toString();
//                 },
//                 clear: function() {
//                   store = {};
//                 },
//                 removeItem: function(key) {
//                   delete store[key];
//                 }
//               };
//     })();
//     Object.defineProperty(window, 'localStorage', { value: localStorageMock });