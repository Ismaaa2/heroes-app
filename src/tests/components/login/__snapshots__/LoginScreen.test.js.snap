// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name of the group debe de mostrarse correctamente 1`] = `
<div
  className="container"
>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
